apps/meteor/app/e2e/server/methods/setUserPublicAndPrivateKeys.ts

```diff
@@ -23,5 +23,8 @@ Meteor.methods<ServerMethods>({
 			private_key: keyPair.private_key,
 			public_key: keyPair.public_key,
 		});
+
+		const subscribedRoomIds = await Rooms.getSubscribedRoomIdsWithoutE2EKeys(userId);
+		await Rooms.addUserIdToE2EEQueueByRoomIds(subscribedRoomIds, userId);
```

Comment:
I like this that we are only adding users after we have ensured that they have their E2EE key pair.

We should also remove users from waiting queue, when they reset their keys again, as they no longer have their keys.

Reply:
Good one!